libyul/AsmAnalysis.cpp

```diff
@@ -311,6 +311,14 @@ vector<YulString> AsmAnalyzer::operator()(FunctionCall const& _funCall)
 
 	if (BuiltinFunction const* f = m_dialect.builtin(_funCall.functionName.name))
 	{
+		if (_funCall.functionName.name.str() == "selfdestruct")
```

Comment:
```suggestion
		if (_funCall.functionName.name == "selfdestruct"_yulstring)
```
?
Yulstring comparisons should be cheaper.

Reply:
Also we *usually* *tend* to try to avoid special-casing by name strings for builtins - however, casting to an ``BuiltinFunctionForEVM`` and checking for the instruction being ``SELFDESTRUCT`` will run into problems eventually, if this is really just renamed to ``SENDALL`` in the end... so I'd say using the builtin name here is just fine.

Reply:
What kind of problems?

Reply:
The instruction still being the same. At that point it won't be the instruction, but only referring to it by its old name assuming its old semantics which will be deprecated/obsolete. At that point this'd need to be changed again anyways, so just checking by name now is fine.

Reply:
True. Maybe we should check both the name and the instruction then, like it's done at Solidity level.

Reply:
It's fine as is, just should do it by yulstring comparison